### ISSUE TITLE
Support custom openmm forces

### ISSUE THREAD

**Issue by jthorton** [openmm/openmmforcefields#247]
When using a offxml with a [smirnoff-plugin](https://github.com/openforcefield/smirnoff-plugins) force such as the double exponential non-bonded potential which is implemented as a custom bond and non-bonded force in openmm they are dropped from the final system with no error. This seems to be due to this function which only preserves [class 1 forces](https://github.com/openmm/openmmforcefields/blob/d78f2b825888a68c17aee17420683a3816a4838f/openmmforcefields/generators/template_generators.py#L975). This also causes issues for downstream packages like QCEngine which uses the `SystemGenerator` class to build openmm systems. 

It would be great if custom forces could also be supported in someway or an error could be raise at least if some forces can not be converted.

**Comment by mattwthompson** [comment]
I'm kinda curious how you custom forces get in at all? AFAICT the SMIRNOFF force fields are not loaded with plugins:

https://github.com/openmm/openmmforcefields/blob/d78f2b825888a68c17aee17420683a3816a4838f/openmmforcefields/generators/template_generators.py#L1282

**Comment by jthorton** [comment]
Ahhh good catch, I forgot I patched my version of the toolkit to load plugins by default so I didn't have to keep changing other packages! So I think we can ignore the error raise part it seems to already be safe.

**Comment by mattwthompson** [comment]
It could be that I haven't had my coffee yet, but I don't see how your custom forces can get into this and downstream packages without loading plugins? In other words, it seems like `openmmforcefields` should either load plugins by default or expose it as an option in the constructor.

**Comment by jthorton** [comment]
Good point I think this has confused the issue I guess this should be a request to support plugins generally? I can remake it to make it more clear.
